Dpsgraph

synopia opened the pull request:
New tab with a dps graph like seen at lailai. Can be disabled in settings.

synopia committed:
DPS Graph

synopia committed:
added death markers

Henjuro commented:
Should be nice to have a dps graph but I think  you need to add your setting to src-electron/util/app-settings.ts too

Henjuro commented:
If I may ask this graph is average dps from the start of the raid to each data point not a rolling window one yes?

Henjuro commented:
I wonder if maybe it would be better to have a rolling window graph.
Graphs are from vykas g3
Like this is 1s interval for datapoints with 10s damage avg (5s before, 5s after)
![image](https://user-images.githubusercontent.com/104998825/227784628-f1c148ed-0dfb-4a52-ad76-7afeed45bd07.png)
vs current 5s interval for datapoints with avg dps since start of raid
![image](https://user-images.githubusercontent.com/104998825/227784888-cfc09d3d-797d-48f3-b9b9-6e57a4a15f2a.png)

also maybe allow for zooming by adding ToolboxComponent, DataZoomComponent and

`    toolbox: {
      left: 'center',
      itemSize: 25,
      top: 55,
      feature: {
        dataZoom: {
          yAxisIndex: 'none'
        },
        restore: {}
      }
    },
    dataZoom: [
      {
        type: 'inside',
        throttle: 50
      }
    ],`

synopia committed:
added setting to app-settings

synopia committed:
Rolling DPS window

synopia committed:
fixed commit

synopia commented:
I modified it to have a fixed number of intervals (100 for now, should be moved to settings) showing the rolling avg for 4 times the interval length. Also refactored to use the same player names functionality used in normal DMG table.
